DHCP ServiceInfo deprecation in 2025.2

jamesonuk opened the pull request:
Resolves https://github.com/mattrayner/pod-point-home-assistant-component/issues/73
An internal class has been moved in HA so it just changing the package.   Looks to be effective from 2025.2 so have bumped minimum version in HACS

jamesonuk committed:
DHCP ServiceInfo deprecation in 2025.2

jamesonuk committed:
Bump pytest-homeassistant-custom-component

Need to bump to 2025.2

jamesonuk committed:
Linting

jamesonuk committed:
Merge branch 'master' of https://github.com/jamesonuk/pod-point-home-assistant-component

jamesonuk committed:
Bump to Python 3.13

mattrayner reviewed (dismissed):
I'll take a look at the tests that are failing later this week and cut a release, unless you are able to resolve them first?

jamesonuk committed:
Fix deprecated imports
Removed in https://github.com/home-assistant/core/pull/131843

jamesonuk committed:
More deprecated constants
https://github.com/home-assistant/core/pull/131799

jamesonuk committed:
And more deprecated constants

jamesonuk committed:
And some more deprecation

jamesonuk commented:
> I'll take a look at the tests that are failing later this week and cut a release, unless you are able to resolve them first?

I was a bit silly and thought it would just a one liner so haven't setup a proper dev environment so I have been a bit lazy and just seeing what GitHub workflow is spitting out....    Can't be many left now....   Probably worth rebasing this branch before merging....

jamesonuk committed:
Wrong update

jamesonuk committed:
reauth flow changes
https://developers.home-assistant.io/blog/2024/11/04/reauth-reconfigure-entry-id/

jamesonuk committed:
fix async issue

jamesonuk commented:
I will try and take a look at some point but it is structural changes now rather than just moving constants.
https://developers.home-assistant.io/blog/2024/11/04/reauth-reconfigure-entry-id/

so looking at 
https://github.com/home-assistant/core/blob/d471de5645b8bf02d500dbddb6b9120578c15c55/homeassistant/config_entries.py#L1200

the result looks to be swallowed up there so he check on results in the test are failing.

jamesonuk committed:
Revert "fix async issue"

This reverts commit da0ec80fb381fe75705a20e41012077b2d99cc0b.

jamesonuk committed:
Revert "reauth flow changes"

This reverts commit ff476cdfe7c8d950eb3305dd2db2b5ba24ab1447.

jamesonuk committed:
Revert pytest-homeassistant-custom-component to earlier version

jamesonuk committed:
Got reauth test to work

jamesonuk commented:
So I did the sensible thing and actually put this on my dev machine and set it up properly.....
Interestingly the HA docs are still outdated https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/config-flow-test-coverage/ but I managed to pull apart what they are doing in some of the internal integrations.

There is one check I commented out (`assert result["type"] == FlowResultType.FORM`) as there is no `type` in here (but I haven't got into the details as I kind of figured this test is slightly moot?

This PR is a mess....   Not sure what you want to about that test above, remove it try and rework it? and what what you to do about PR.    Happy to clean this up and do a force push or to close this PR and create a new clean one

mattrayner reviewed (approved):
Thanks so much for updating this, I'll cut a new release of this today
